### PR TITLE
update search ux/ui [ready for review]

### PR DESCRIPTION
**Pull request by jdamiba**
closes https://github.com/plotly/graphing-library-docs/issues/7

Work plan:

- [x] query algolia backend using instantsearch.js
- [x] move search box to header
- [x] update "no results found" template with google site search of plot.ly
- [x] create modal UI
  - [x] toggle modal on/off with search input
  - [x] click outside toggles modal off 
  - [x] get hyperlink URLs to render correctly
  - [x] style modal
- [x] turn on multi-index search
    - [x] refactor modal UI to clearly delineate results per index

### CODE REVIEW

**Comment by jdamiba** [comment]
Screenshots:

Screen Shot 2020-03-03 at 2 24 57 PM
Screen Shot 2020-03-03 at 2 24 39 PM
Screen Shot 2020-03-03 at 2 24 33 PM

File: _includes/layouts/_header-main.html [hunk 1]
```diff
@@ -9,6 +9,13 @@
         <div class="--wrap-right">
             <nav class="--nav-meta" role="navigation">
                 <ul>
+                    {% unless page.permalink == "/api/" %}
```
Comment (nicolaskruchten) [inline _includes/layouts/_header-main.html]: this should probably only be on the 4 target languages right? not e.g. MATLAB or Julia etc

File: all_static/javascripts/algolia/algolia.js [hunk 2]
```diff
@@ -9,8 +9,13 @@ $(function(config) {
   var schemaSearchResults = document.getElementById('schema-search-results');
   var modalContent = document.getElementById('modal-content');
 
-  var emptyResult = '<div class="text-center">No results found matching <strong>{{query}}</strong> Try the search on <a target="_blank" href="https://www.google.com/search?q=site%3Aplot.ly+{{query}}">Google</a>.</div>';
+  var lang = window.plotly_doc_language;
+  if (lang == "plotly_js"){
+    lang = "javascript"
+  }
 
+  var emptyResult = '<div class="text-center"><br><br>No results found matching <strong>{{query}}</strong>.<br><br> Click here to <a class="algolia__result-link" target="_blank" href="https://www.google.com/search?q=site%3Aplot.ly+{{query}}">search for "plotly ' + lang + ' {{query}}" on Google</a>.</div>';
```
Comment (nicolaskruchten) [inline all_static/javascripts/algolia/algolia.js]: need to fix the search query to match what it says in quotes :)

File: _includes/layouts/_header-main.html (hunk 1 above)
Comment (jdamiba) [reply _includes/layouts/_header-main.html]: Changed the conditional to target correct set of pages.

File: all_static/javascripts/algolia/algolia.js (hunk 2 above)
Comment (jdamiba) [reply all_static/javascripts/algolia/algolia.js]: fixed query to match text.

**Comment by jdamiba** [comment]
@nicolaskruchten This PR is ready for review.

File: all_static/css/main.css [hunk 3]
```diff
@@ -9279,3 +9279,92 @@ pre, code, .hljs {
...
+}
+
+.algolia__result-link{
+  color: #2391fe !important;
+}
+
+.modal-body{
+  background: white;
+  padding: 0 !important;
+  max-height: 75vh;
```
Comment (nicolaskruchten) [inline all_static/css/main.css]:
```suggestion
  max-height: calc(100vh - 200px);
```

File: all_static/css/main.css [hunk 4]
```diff
@@ -9279,3 +9279,92 @@ pre, code, .hljs {
...
+  padding-bottom: 20px;
+}
+
+.modal-header{
+  background: #fff;
+}
+
+.modal > .modal-dialog{
+  width: 1050px;
+  height: 700px;
```
Comment (nicolaskruchten) [inline all_static/css/main.css]: let's get rid of this line here, which is adding an unnecessary scrollbar.

**Comment by nicolaskruchten** [comment]
Last two CSS changes then merge! 💃

**Pull request merged**